xen-netback: remove unconditional pull_skb_tail on vif tx path

Unconditionally pulling 128 bytes into the linear region is not required

The checksum code pulls as much packet header data as required into the linear region.

The unconditional pull results in wasted memcpy overhead and touching frag data 
unnecessarily.
diff --git a/drivers/net/xen-netback/netback.c b/drivers/net/xen-netback/netback.c
index 6dd159e..21c9a71 100644
--- a/drivers/net/xen-netback/netback.c
+++ b/drivers/net/xen-netback/netback.c
@@ -1748,11 +1748,6 @@ static int xenvif_tx_submit(struct xenvif *vif)
 			}
 		}
 
-		if (skb_is_nonlinear(skb) && skb_headlen(skb) < PKT_PROT_LEN) {
-			int target = min_t(int, skb->len, PKT_PROT_LEN);
-			__pskb_pull_tail(skb, target - skb_headlen(skb));
-		}
-
 		skb->dev      = vif->dev;
 		skb->protocol = eth_type_trans(skb, skb->dev);
 		skb_reset_network_header(skb);
